presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -343,7 +353,8 @@ public LookupSource createLookupSource(List<Integer> joinChannels, Optional<Inte
                 LookupSource lookupSource = lookupSourceFactory.createLookupSource(
                         valueAddresses,
                         ImmutableList.<List<Block>>copyOf(channels),
-                        hashChannel);
+                        hashChannel,
+                        createJoinFilterFunctionVerifier(filterFunction, ImmutableList.<List<Block>>copyOf(channels)));
```

Comment:
Yeah. This is also true for `copyOf` 2 lines above this one ;). For some reason IntelliJ adds those unnecessary specifications sometimes and does not emit warning. 
Removed both.